test/unit/test_multiple_schema_versions.py

```diff
@@ -0,0 +1,73 @@
+import pytest
+import serialized_data_interface as sdi
+import yaml
+from ops.charm import CharmBase
+from ops.testing import Harness
+from jsonschema.exceptions import ValidationError
+
+
+class ProvideCharm(CharmBase):
+    def __init__(self, *args):
+        super().__init__(*args)
+        with open("test/unit/test_schema.yaml", "r") as schema:
+            self.interface = sdi.SerializedDataInterface(
+                self,
+                "app-provides",
+                yaml.safe_load(schema),
+                {"v1", "v2"},
+                "provides",
+            )
+
+
+def test_send_data_multiple_versions():
+    harness = Harness(
+        ProvideCharm,
+        meta="""
+        name: test-app
+        provides:
+            app-provides:
+                interface: serialized-data
+        """,
+    )
+    harness.set_leader(True)
+
+    rel_id1 = harness.add_relation("app-provides", "foo")
+    harness.add_relation_unit(rel_id1, "foo/0")
+    harness.update_relation_data(
+        rel_id1,
+        "foo",
+        {"_supported_versions": "- v1"},
+    )
+
+    rel_id2 = harness.add_relation("app-provides", "bar")
+    harness.add_relation_unit(rel_id2, "bar/0")
+    harness.update_relation_data(
+        rel_id2,
+        "bar",
+        {"_supported_versions": "- v2"},
+    )
+
+    harness.begin()
+    data = {
+        "service": "my-service",
+        "port": 4242,
+        "access-key": "my-access-key",
+        "secret-key": "my-secret-key",
+    }
+    harness.update_relation_data(rel_id1, "foo", {"data": yaml.dump(data)})
+    harness.update_relation_data(rel_id2, "bar", {"data": yaml.dump(data)})
+
+    harness.charm.interface.send_data(data, "foo")
+    harness.charm.interface.send_data({"foo": "bar"}, "bar")
```

Comment:
for clarity, it might help to change one of the `bar`'s here.  If I understand correctly, we are sending data to `app_name=bar` on the key `foo` with content `bar`, and that this schema that the app `bar` uses also has `bar` in the schema.  There's just a lot of `bar` :)

Reply:
@ca-scribner good call, updated